dynawo/sources/ModelicaCompiler/Scripts_OMC_1_13_2/factory.py

```diff
@@ -2600,11 +2569,47 @@ def get_list_for_evalcalculatedvars(self):
     # @return
     def prepare_for_evalcalculatedvari(self):
         index = 0
-        for var in self.list_calculated_vars:
-            expr = self.dic_calculated_vars_values[var.get_name()]
+        trans = Transpose(self.reader.auxiliary_vars_to_address_map, self.reader.derivative_residual_vars + self.reader.assign_residual_vars)
+        ptrn_vars = re.compile(r'x\[(?P<varId>[0-9]+)\]')
+        self.list_for_evalcalculatedvari.append("  data->simulationInfo->discreteCall = 1;\n")
```

Comment:
At this stage, I'm not very sure we should place discreteCall to 1. If I'm right, we can call evalcalculatedvari during an evaluation of f (in case it is a connector). In this situation, for example for the macro RELATIONHYSTERESIS, if discreteCall is set to 1, then you evaluate the comparator to know what is the current f and you potentially change them during the Newton iterations.
It is something we don't want to do anymore but change them during the evalMode call. Thus it should be in the evalMode part that we change the equation on calculated variables.

Reply:
Current fix: whenever RELATIONHYSTERESIS is present in the code used to calculate the variable, it cannot become a calculated variable. We could add later this possibility by recalculating the variables in evalMode and adapting the connectors.